optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/tenant/NewTenantForm.java

```diff
@@ -0,0 +1,168 @@
+package org.optaplanner.openshift.employeerostering.gwtui.client.tenant;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import elemental2.dom.HTMLButtonElement;
+import elemental2.dom.HTMLDivElement;
+import elemental2.dom.HTMLInputElement;
+import elemental2.dom.MouseEvent;
+import org.gwtbootstrap3.extras.select.client.ui.Option;
+import org.gwtbootstrap3.extras.select.client.ui.Select;
+import org.jboss.errai.ui.client.local.api.elemental2.IsElement;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.ForEvent;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.optaplanner.openshift.employeerostering.gwtui.client.common.FailureShownRestCallback;
+import org.optaplanner.openshift.employeerostering.gwtui.client.common.LocalDatePicker;
+import org.optaplanner.openshift.employeerostering.gwtui.client.popups.FormPopup;
+import org.optaplanner.openshift.employeerostering.gwtui.client.popups.PopupFactory;
+import org.optaplanner.openshift.employeerostering.shared.roster.RosterState;
+import org.optaplanner.openshift.employeerostering.shared.tenant.Tenant;
+import org.optaplanner.openshift.employeerostering.shared.tenant.TenantRestServiceBuilder;
+
+@Templated
+public class NewTenantForm implements IsElement {
+
+    @Inject
+    @DataField("tenant-name-text-box")
+    private HTMLInputElement tenantName;
+
+    @Inject
+    @DataField("schedule-start-date")
+    private LocalDatePicker scheduleStartDate;
+    
+    @Inject
+    @DataField("schedule-start-weekday")
+    private HTMLDivElement scheduleStartWeekday;
+
+    @Inject
+    @DataField("publish-notice")
+    private HTMLInputElement publishNotice;
+
+    @Inject
+    @DataField("publish-length")
+    private HTMLInputElement publishLength;
```

Comment:
This field should be read-only according to the feature description: https://issues.jboss.org/browse/PLANNER-1085.

@ge0ffrey is this requirement still valid?